rali/rali/source/tf_meta_data_reader.cpp

```diff
@@ -0,0 +1,197 @@
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+THE SOFTWARE.
+*/
+
+#include "tf_meta_data_reader.h"
+#include <iostream>
+#include <utility>
+#include <algorithm>
+#include<fstream>
+#include <string>
+#include <stdint.h>
+#include <google/protobuf/message_lite.h>
+#include "example.pb.h"
+#include "feature.pb.h"
+
+using namespace std;
+
+void TFMetaDataReader::init(const MetaDataConfig &cfg)
+{
+    _path = cfg.path();
+    _output = new LabelBatch();
+    _last_rec = false;
+}
+
+bool TFMetaDataReader::exists(const std::string& _image_name)
+{
+    return _map_content.find(_image_name) != _map_content.end();
+}
+
+void TFMetaDataReader::add(std::string _image_name, int label)
+{
+    pMetaData info = std::make_shared<Label>(label);
+    if(exists(_image_name))
+    {
+        WRN("Entity with the same name exists")
+        return;
+    }
+    _map_content.insert(pair<std::string, std::shared_ptr<Label>>(_image_name, info));
+}
+
+void TFMetaDataReader::lookup(const std::vector<std::string> &_image_names)
+{
+    if(_image_names.empty())
+    {
+        WRN("No image names passed")
+        return;
+    }
+    if(_image_names.size() != (unsigned)_output->size())   
+        _output->resize(_image_names.size());
+
+    for(unsigned i = 0; i < _image_names.size(); i++)
+    {
+        auto _image_name = _image_names[i];
+        auto it = _map_content.find(_image_name);
+        if(_map_content.end() == it)
+            THROW("ERROR: Given name not present in the map"+ _image_name )
+        _output->get_label_batch()[i] = it->second->get_label();
+    }
+
+}
+
+void TFMetaDataReader::print_map_contents()
+{
+    std::cerr << "\nMap contents: \n";
+    for (auto& elem : _map_content) {
+        std::cerr << "Name :\t " << elem.first << "\t ID:  " << elem.second->get_label() << std::endl;
+    }
+}
+
+MetaDataReader::Status TFMetaDataReader::read_record(std::ifstream &file_contents, uint file_size, std::vector<std::string> &_image_name)
+{
+    auto ret = MetaDataReader::Status::OK;
+    uint length;
+    length = file_contents.tellg();
+    size_t uint64_size, uint32_size;
+    uint64_t data_length;
+    uint32_t length_crc, data_crc;
+    uint64_size = sizeof(uint64_t); 
+    uint32_size = sizeof(uint32_t); 
+    char * header_length = new char [uint64_size];
+    char * header_crc = new char [uint32_size];
+    char * footer_crc = new char [uint32_size];
+    file_contents.read(header_length, uint64_size);
+    if(!file_contents)
+        THROW("TFMetaDataReader: Error in reading TF records")
+    file_contents.read(header_crc, uint32_size);
+    if(!file_contents)
+        THROW("TFMetaDataReader: Error in reading TF records")
+    memcpy(&data_length, header_length, sizeof(data_length));
+    memcpy(&length_crc, header_crc, sizeof(length_crc));
+    if(length + data_length + 16 == file_size){
+        _last_rec = true;
+    }
+    char *data = new char[data_length];
+    file_contents.read(data,data_length);
```

Comment:
Codacy found an issue: [Check buffer boundaries if used in a loop including recursive loops (CWE-120, CWE-20).  ](https://app.codacy.com/gh/GPUOpen-ProfessionalCompute-Libraries/MIVisionX/pullRequest?prid=5763058)

Reply:
This is fstream read. We are evaluating the read status in the next line.
if(!file_contents)
      THROW("TFMetaDataReader: Error in reading TF records")
This can be ignored.